modules/preview-react/side-panel/lib/SidePanelToggleButton.tsx

```diff
@@ -0,0 +1,78 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import * as React from 'react';
+import {createComponent, ExtractProps, useIsRTL} from '@workday/canvas-kit-react/common';
+import {css, jsx} from '@emotion/react';
+import {TertiaryButton} from '@workday/canvas-kit-react/button';
+import {space} from '@workday/canvas-kit-react/tokens';
+import {transformationImportIcon} from '@workday/canvas-system-icons-web';
+import {Tooltip} from '@workday/canvas-kit-react/tooltip';
+import {SidePanelContext} from './hooks';
+
+export interface SidePanelToggleButtonProps extends ExtractProps<typeof TertiaryButton, never> {
+  /**
+   * The tooltip text to expand the side panel
+   */
+  tooltipTextExpand?: string;
+  /**
+   * The tooltip text to collapse the side panel
+   */
+  tooltipTextCollapse?: string;
+}
+
+/**
+ * A toggle button styled specifically for the side panel container.
+ */
+export const SidePanelToggleButton = createComponent('button')({
```

Comment:
Just curious why this is `createComponent('button')` and not `createComponent(TertiaryButton)`. I noticed the docs for SidePanel.ToggleButton mention the props extend from `button` and not `TertiaryButton`:

<img width="713" alt="Screenshot 2023-02-28 at 6 13 57 PM" src="https://user-images.githubusercontent.com/146020/222027512-a852b732-8428-4ec4-b9dd-f3e867e36878.png">

I was expecting something similar to ActionBar.Item which mentions its props extending from SecondaryButton:

<img width="769" alt="image" src="https://user-images.githubusercontent.com/146020/222028063-457badf6-631c-4388-bf41-d9d6f06e7ddf.png">

I did notice ActionBar.Item uses `createSubcomponent` whereas SidePanel.ToggleButton uses `createComponent`.

Reply:
The element vs component depends on who owns the styling.

- If the `as` component owns the styling, the default should be a component
- If the `as` component does not own the styling, the default should be an element

In the case of the `Popup.Target`, or any of the other `*.Target` subcomponents, only the behavior is owned by the subcomponent. The component passed in owns its own styling. 99% of cases the subcomponent owns styling.

In the case of `SidePanel.ToggleButton`, the original toggle button owns styling. The original `SidePanel.ToggleButton` accepted an `as` prop, but that didn't change styling: https://github.com/Workday/canvas-kit/blob/6c4b62faa6668a57c6d77579b411fea31ecc02a3/modules/react/side-panel/lib/SidePanel.tsx#L140

Changing the default rendered component to `SecondaryButton` would change the API interface which would be a breaking change.

I was debating how to better inform the component docs what base element was rendered. Most of the time, the "Style Component" or "Layout Component" takes on this role, but that only works by detecting symbol declaration file names and not detecting the base element rendered. I can't think of a better solution. Does passing an `as` change the _element_ rendered or the _component_ rendered? How it documents now answers this question. Perhaps the "Style Component" or "Layout Component" should be updated to detect the root JSX node rendered instead?

```suggestion
export const SidePanelToggleButton = createComponent('button')({
```